pkg/metamorph/client_test.go

```diff
@@ -441,3 +449,323 @@ func TestClient_SubmitTransactions(t *testing.T) {
+		{"double_spend_attempted", "testTxID11", metamorph_api.Status_DOUBLE_SPEND_ATTEMPTED, "DOUBLE_SPEND_ATTEMPTED", "DOUBLE_SPEND_ATTEMPTED"},
+		{"rejected", "testTxID12", metamorph_api.Status_REJECTED, "REJECTED", "REJECTED"},
+		{"mined", "testTxID13", metamorph_api.Status_MINED, "MINED", "MINED"},
+		{"transaction not found", "missingTxID", metamorph_api.Status_UNKNOWN, "", metamorph.ErrTransactionNotFound.Error()},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			// Given
+			mockClient := &apiMocks.MetaMorphAPIClientMock{
+				GetTransactionStatusFunc: func(ctx context.Context, req *metamorph_api.TransactionStatusRequest, opts ...grpc.CallOption) (*metamorph_api.TransactionStatus, error) {
+					if tc.expectedErrStr != "" {
+						return nil, errors.New(tc.expectedErrStr)
+					}
+					return &metamorph_api.TransactionStatus{
+						Txid:        tc.txID,
+						Status:      tc.status,
+						BlockHash:   "sampleBlockHash",
+						BlockHeight: 100,
+					}, nil
+				},
+			}
+
+			now := time.Now().Unix()
+			client := metamorph.NewClient(mockClient, metamorph.WithNow(func() time.Time { return time.Unix(now, 0) }))
+
+			// When
+			status, err := client.GetTransactionStatus(context.Background(), tc.txID)
+
+			// Then
+			if tc.expectedErrStr != "" {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), tc.expectedErrStr)
+				require.Nil(t, status)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, &metamorph.TransactionStatus{
+					TxID:        tc.txID,
+					Status:      tc.expectedStatus,
+					BlockHash:   "sampleBlockHash",
+					BlockHeight: 100,
+					Timestamp:   now,
+				}, status)
+			}
+
+			require.Equal(t, 1, len(mockClient.GetTransactionStatusCalls()), "Unexpected number of GetTransactionStatus calls")
+		})
+	}
+}
+
+func TestClient_Health(t *testing.T) {
+	tt := []struct {
+		name           string
+		mockError      error
+		expectedErrStr string
+	}{
+		{
+			name:           "success - health check passes",
+			mockError:      nil,
+			expectedErrStr: "",
+		},
+		{
+			name:           "error - health check fails",
+			mockError:      errors.New("health check error"),
+			expectedErrStr: "health check error",
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			// Given
+			mockClient := &apiMocks.MetaMorphAPIClientMock{
+				HealthFunc: func(ctx context.Context, in *emptypb.Empty, opts ...grpc.CallOption) (*metamorph_api.HealthResponse, error) {
+					return &metamorph_api.HealthResponse{}, tc.mockError
+				},
+			}
+
+			client := metamorph.NewClient(mockClient)
+
+			// When
+			err := client.Health(context.Background())
+
+			// Then
+			if tc.expectedErrStr == "" {
+				require.NoError(t, err)
+			} else {
+				require.ErrorContains(t, err, tc.expectedErrStr)
+			}
+
+			require.Equal(t, 1, len(mockClient.HealthCalls()), "Unexpected number of Health calls")
+		})
+	}
+}
+
+func TestGetTransactions(t *testing.T) {
```

Comment:
This tests the same function as `TestClient_GetTransactions`
Is this a duplicate?
Merge the test cases into one unit test